### PR TITLE
Add crossDacMergeLatestTag.sh and docs

### PR DESCRIPTION
**Pull request by sdmaclea**

### CODE REVIEW

**Comment by sdmaclea** [comment]
This is draft quality.  I would appreciate a quick review.  Documentation for coherency and the script for stupid errors.  I will try to hack this to test it locally, but I really haven't started testing it yet.

File: README-CrossDac.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+### Cross OS DAC Readme
+
+This repo is a fork of the primary CoreCLR release/3.1 branch.
+
+It exists to implement the Windows version of the DAC & DBI libraries
+which we use for debugging. It enables examination of Linux core dumps
```
Comment (josalem) [inline README-CrossDac.md:6]:
```suggestion
which we use for debugging and enables examination of Linux core dumps
```

File: README-CrossDac.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+For each of these servicing releases we need to release a new cross OS
+DAC/DBI set.
+
+##### Release Process
+
+Assumes the user has a clone of the dotnet/coreclr repo.
+
+In the afternoon when a servicing release has occurred.
+- Checkout this release/3.1-crossdac branch of the repo.
```
Comment (josalem) [inline README-CrossDac.md]:
```suggestion
- Checkout the release/3.1-crossdac branch of the repo.
```

File: README-CrossDac.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+- Run ./crossDacMergeLatestTag.sh <upstream>
```
Comment (josalem) [inline README-CrossDac.md:28]: An example invocation would be helpful here

File: README-CrossDac.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+- Create a PR to merge from <upstream>:release/3.1-crossdac-<latestTag>
+  to dotnet/coreclr:release/3.1-crossdac
+  - Review PR looking for changes which could potentially break the Cross
+    DAC/DBI. These would typically be layout changes to structures or the
+    addition of HOST_* conditional code
+  - In the absence of risky changes, merge the commit.
+  - If there are riskier changes extra eyes might be useful to see if the
+    changes are breaking. (Not really expected.)
+- When the PR is merged an internal trial build will be triggered. See
+  https://dev.azure.com/dnceng/internal/_build?definitionId=244&_a=summary.
```
Comment (josalem) [inline README-CrossDac.md:43]: This URL points to an internal AzDO instance and most likely won't be helpful to anyone outside msft. I would leave it out.

**Review by josalem — COMMENTED** [review]
Only some grammatical nits.  Otherwise I think this looks good 👍.  Let me know when you want me to go through it again.  The script looks good as well.

File: README-CrossDac.md (hunk 1 above)
Comment (sdmaclea) [reply README-CrossDac.md:6]: Feels like this introduces a run-on sentence.

File: README-CrossDac.md (hunk 4 above)
Comment (sdmaclea) [reply README-CrossDac.md:43]: I don't know.  I find it difficult to find these pipelines. We could move this tribal knowledge into the team OneNote.  But then this whole document could/should go there instead.

**Comment by sdmaclea** [comment]
I just tested the script.  Here is my hack PR it created (from my dummy local v3.1.85 tag). https://github.com/dotnet/coreclr/pull/28168.

**Comment by sdmaclea** [comment]
I plan to merge this soon unless there is more feedback

**Review by josalem — APPROVED** [review]

**Pull request merged**